src/views/chat/chat.jsx

```diff
@@ -0,0 +1,215 @@
+  const [userMessages, setUserMessages] = useState([]);
+  const [newMessage, setNewMessage] = useState('');
+  const [userAvatar, setUserAvatar] = useState('');
+  const chatMessagesRef = useRef(null);
+  const [showChats, setShowChats] = useState(false);
+  const [showChatButton, setShowChatButton] = useState(true);
+  const [unreadCount, setUnreadCount] = useState(0);
+  const inputRef = useRef(null);
+
+  const room = 'chatbot';
+  const user = getUserInfo();
+
+  if (!user) return;
+
+  socket.emit('join_room', room);
+
+  const handleChatButtonClick = () => {
+    setShowChats(false);
+    setShowChatButton(true);
+    setUnreadCount(0);
+  };
+
+  socket.on('username', ({ name, avatar }) => {
+    setUserName(name);
+    setUserAvatar(avatar);
+  });
+
+  const userId = user.payload?.id;
+
+  const handleTypingStart = () => {
+    socket.emit('typing', { user: userName, room, isTyping: true });
+  };
+
+  const handleSendMessage = () => {
+    if (newMessage.trim() === '') {
+      toast.error('Oops! Your message is empty \u{1F614}');
+      return;
+    }
+    socket.emit('send-chat-message', {
+      message: newMessage,
+      username: userName,
+      avatar: userAvatar,
+      date: new Date(),
+    });
+
+    if (inputRef.current) {
+      inputRef.current.value = '';
+    }
+    setNewMessage('');
+  };
+
+  useEffect(() => {
+    socket.on('connect', () => {
+      setIsSocketConnected(true);
+      console.log('socket connected');
+    });
+
+    socket.on('disconnect', () => {
+      setIsSocketConnected(false);
+    });
+
+    socket.on('istyping', (data) => {
+      setTypingText(`@${data.user} is typing ...`);
+      setTimeout(() => {
+        setTypingText('');
+      }, 3000);
+    });
+
+    socket.on('all-messages', (messages) => {
+      setUserMessages(messages);
+    });
+
+    socket.on('chat-message', (message) => {
+      setUserMessages((prevMessages) => [...prevMessages, message]);
+      if (!showChats) {
+        setUnreadCount((prevCount) => prevCount + 1);
+      }
+    });
+
+    return () => {
+      socket.off('istyping');
+      socket.off('chat-message');
+    };
+  }, [socket]);
+
+  useEffect(() => {
+    const chatMessagesContainer = chatMessagesRef.current;
+    if (chatMessagesContainer) {
+      chatMessagesContainer.scrollTop = chatMessagesContainer.scrollHeight;
+    }
+  }, [userMessages]);
+
+  return (
+    <>
+      <div className={`chat-container ${!showChats ? 'form-chat-hidden' : ''}`}>
+        <div className="chat-header">
+          <div className="chat-header-title">Chat</div>
+          <button onClick={handleChatButtonClick} className="close-button">
+            X
+          </button>
+        </div>
+        <div className="chat-messages" ref={chatMessagesRef}>
+          {userMessages.map((message) => {
+            const isSender = message.senderId === userId;
+
+            return (
+              <div
+                key={message.id}
```

Comment:
If we don't have userMessages this will throw error first check if there is message